### ISSUE TITLE
Handling subdirectories

### ISSUE THREAD

**Issue by nelstrom** [k-takata/minpac#33]
Here are a couple of examples of repositories that include a vim plugin (as well as plugins for other tools):

* [sonph/onehalf](https://github.com/sonph/onehalf)
* [nsf/gocode](https://github.com/nsf/gocode)

Do you have any advice on how to install these plugins using minpac?

**Comment by nelstrom** [comment]
vim-plug has an `rtp` option which is useful in this scenario. e.g. you can use:

    Plug 'nsf/gocode', { 'rtp': 'vim' }

That will check out the entire `nsf/gocode` repository, then set Vim's runtimepath to include the `vim` directory within that repository. I'm guessing that this approach won't work here, since minpac doesn't manage Vim's runtimepath.

To make this work for minpac, I think it would be necessary to only checkout the `vim` subdirectory of the repository. Apparently Git can do this using [sparse checkouts](https://stackoverflow.com/a/2303645/128850).

**Comment by k-takata** [comment]
Hmm, does sparse checkout strip directory structure?
I think it checks out only specified files/directories but the directory structure is kept.

One possible workaround is using another package and create a symlink in it.

```shell
mkdir -p ~/.vim/pack/bundle/start
cd ~/.vim/pack
ln -s minpac/start/gocode/vim bundle/start/gocode
```

(not tested)

**Comment by nelstrom** [comment]
You're right about sparse checkout: it does maintain the directory structure of the original repo. My mistake!

The symlink idea makes sense. I suppose you could get away with putting both the full repository and the symlink in the same package? e.g.:

    cd ~/.vim/pack/minpac/start
    git clone https://github.com/nsf/gocode.git gocode-full
    ln -s gocode-full/vim gocode

Here, I'm using `gocode-full` as the name of the full repository, and `gocode` as the name for the symlink to the `vim` subdirectory within.

**Comment by k-takata** [comment]
In that case, `minpac#clean()` will remove `gocode`.

**Comment by nelstrom** [comment]
Oh yeah, I hadn't thought of that.

I was wondering whether you might consider adding an option to `minpac#add()` so that you could do something like this?

    call minpac#add('nsf/gocode', { 'subdir': 'vim' })

I'm not sure if `subdir` is the right choice of name for that key, but you get the idea. (The `rtp` option used by vim-plug and vundle would be a misnomer here) If a plugin is registered with this option, it would become a special case when running `minpac#update()` (and potentially also when running `minpac#clean()`).

Is it worth adding all of that extra code just to handle these special cases? I'm not sure, especially since examples like [sonph/onehalf](https://github.com/sonph/onehalf) and [nsf/gocode](https://github.com/nsf/gocode) are uncommon. Alternatively, it might be worthwhile adding something to the documentation to describe how to install these kinds of plugin using minpac. If that's your preferred option, I can help.

**Comment by cilindrox** [comment]
Just run into this with both [nginx] and [docker] repos

[nginx]: https://github.com/nginx/nginx/tree/master/contrib/vim
[docker]: https://github.com/moby/moby/tree/master/contrib/syntax/vim

**Comment by jgouly** [comment]
I also want this feature. https://github.com/l04m33/vlime needs a subdirectory for the rtp.

**Comment by tregusti** [comment]
Isn't it possible to only change the dir to clone into to a special dir for "full" repos, and then just symlink the `dir` option value.

```sh
mkdir -p ~/.vim/pack/minpac/full
cd ~/.vim/pack/minpac/full
git clone https://github.com/nsf/gocode.git
cd ~/.vim/pack/minpac/start
ln -s ../full/gocode gocode
```

**Comment by jsatk** [comment]
I too am having the same issue with https://github.com/l04m33/vlime.

**Comment by jsatk** [comment]
@k-takata I'd be willing to help add in the ability to set the runtimepath similar to how vim-Plug does it (mentioned here: https://github.com/k-takata/minpac/issues/33#issuecomment-315021111). Are you actively working on this? Or is this in progress in any way?

**Comment by k-takata** [comment]
Sorry for being late.
I implemented this in the `devel` branch. (#97)

**Comment by cilindrox** [comment]
Thanks for this @k-takata
